backend/infrahub/core/schema/schema_branch.py

```diff
@@ -224,8 +224,22 @@ def update(self, schema: SchemaBranch) -> None:
         #     else:
         #         del self.generics[item_kind]
 
-    def validate_update(self, other: SchemaBranch, enforce_update_support: bool = True) -> SchemaUpdateValidationResult:
-        diff = self.diff(other=other)
+    def validate_node_deletions(self, diff: SchemaDiff) -> None:
+        """Given a diff, check if a deleted node is still used in relationships of other nodes."""
+        for schema_name in diff.removed:
+            for name in self.all_names:
```

Comment:
I think you could do this with one loop through `self.all_names` if you replaced the `for schema_name in diff.removed` with `removed_schema_names = set(diff.removed.keys())`